frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,201 @@
+        });
+        return;
+      }
+      if (!loaded) {
+        return;
+      }
+
+      const state = {
+        items: {},
+        title: {},
+      };
+
+      const noStorageClass = 'No storage class';
+      state.items[noStorageClass] = { kindLabel: '', name: noStorageClass };
+      _.each(resources, ({ data }, kindLabel) => {
+        _.reduce(
+          data,
+          (acc, resource) => {
+            acc[resource.metadata.name] = {
+              kindLabel,
+              name: resource.metadata.name,
+              description: resource.metadata.annotations ? resource.metadata.annotations.description : '',
+              default: resource.metadata.annotations ? resource.metadata.annotations['storageclass.beta.kubernetes.io/is-default-class'] === 'true' ? this.setState({defaultClass:resource.metadata.name}, ()=> {
+                return true;
+              }) : false : '',
+              accessMode: resource.metadata.annotations ? resource.metadata.annotations['storage.alpha.openshift.io/access-mode'] : '',
+              provisioner: resource.provisioner,
+              type: resource.parameters ? resource.parameters.type : '',
+              zone: resource.parameters ? resource.parameters.zone : '',
+            };
+            return acc;
+          },
+          state.items
+        );
+      });
+
+      const { defaultClass } = this.state;
+      if (defaultClass) {
+        // No Storage Class is no longer an option if there is a default
+        delete state.items[noStorageClass];
+      }
+
+      if (!this.props.loaded || !selectedKey || selectedKey === noStorageClass) {
+        state.title = <span className="text-muted">{nextProps.placeholder}</span>;
+      }
+
+      if (selectedKey && selectedKey !== noStorageClass) {
+        const item = state.items[selectedKey];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      if (!selectedKey && defaultClass) {
+        const item = state.items[defaultClass];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      this.setState(state);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+      if (_.isEqual(this.state, nextState)) {
+        return false;
+      }
+      return true;
+    }
+
+    autocompleteFilter = (text, item) => fuzzy(text, item.props.name);
+
+    getTitle = (kindLabel, name) => {
+      return kindLabel ? <ResourceName kind={kindLabel} name={name} /> : <span >{name}</span>;
+    }
+
+    onChange = key => {
+      const { name, kindLabel, accessMode } = _.get(this.state, ['items', key], {});
+      this.setState({
+        selectedKey: key,
+        title: this.getTitle(kindLabel, name),
+      });
+      this.props.onChange(name, accessMode);
+    };
+
+    render() {
+      const { placeholder, id, loaded } = this.props;
+      const items = {};
+      const sortedItems = _.keys(this.state.items).sort();
```

Comment:
It'd be better to do this in component will receive props than in render. Sorting is expensive.

Reply:
Funny thing is the Dropdown component does not have a sortedItemKeys property so I removed it.  It just takes the items in the order they arrive.

Reply:
We have an issue to track this, but the Dropdown will use key insertion order. So you still should be able to keep it sorted. (There are some quirks with values that parse as numbers in Chrome, but otherwise it works.)